Service/Category/BatchQueueProcessor.php

```diff
@@ -43,7 +44,9 @@ public function processBatch(int $storeId, ?array $entityIds = null): void
             return;
         }
 
-        $this->rebuildAffectedProducts($storeId);
+        if (count($this->affectedProductIds) > 0) {
+            $this->rebuildAffectedProducts($storeId);
```

Comment:
Do the `affectedProductIds` get reset after this? Also do we have to maintain two separate sets of IDs? In both this class and the class property in the `Algolia\AlgoliaSearch\Model\Indexer\Category`? If so a single source of truth might be better...